File: app/src/main/java/com/zighter/zighterandroid/util/Optional.java
package com.zighter.zighterandroid.util;

public class Optional<T> {

    private final T value;

    private Optional(T value) {
        this.value = value;
    }

    public boolean isPresent() {
        return value != null;
    }

    public T get() {
        return value;
    }

    public static <T> Optional<T> of(T value) {
        return new Optional<T>(value);
    }

}

File: app/src/main/java/com/zighter/zighterandroid/util/Optional.kt
package com.zighter.zighterandroid.util

class Optional<T> private constructor(private val value: T?) {

    val isPresent: Boolean
        get() = value != null

    fun get(): T? {
        return value
    }

    companion object {
        fun <T> of(value: T): Optional<T> {
            return Optional(value)
        }
    }
}